Source/WebCore/platform/network/BlobRegistryImpl.cpp

```diff
@@ -261,19 +265,31 @@ void BlobRegistryImpl::registerInternalBlobURLForSlice(const URL& url, const URL
     addBlobData(url.string(), WTFMove(newData));
 }
 
-void BlobRegistryImpl::unregisterBlobURL(const URL& url, const std::optional<WebCore::SecurityOriginData>&)
+void BlobRegistryImpl::unregisterBlobURL(const URL& url, const std::optional<WebCore::SecurityOriginData>& topOrigin)
 {
     ASSERT(isMainThread());
-    if (m_blobReferences.remove(url.string()))
-        m_blobs.remove(url.string());
+    ASSERT(BlobURL::isInternalURL(url) || topOrigin);
+    auto& urlKey = url.string();
+    if (m_isPartitioningEnabled && m_blobs.contains(urlKey) && topOrigin && topOrigin != m_allowedBlobURLTopOrigins.get(urlKey)) {
+        RELEASE_LOG(Network, "Rejecting unregistering blob URL with incorrect top origin.");
+        return;
+    }
+    if (m_blobReferences.remove(urlKey)) {
+        m_blobs.remove(urlKey);
+        m_allowedBlobURLTopOrigins.remove(urlKey);
+    }
 }
 
-BlobData* BlobRegistryImpl::getBlobDataFromURL(const URL& url, const std::optional<SecurityOriginData>&) const
+BlobData* BlobRegistryImpl::getBlobDataFromURL(const URL& url, const std::optional<SecurityOriginData>& topOrigin) const
 {
     ASSERT(isMainThread());
-    if (url.hasFragmentIdentifier())
-        return m_blobs.get<StringViewHashTranslator>(url.viewWithoutFragmentIdentifier());
-    return m_blobs.get(url.string());
+    auto urlKey = url.stringWithoutFragmentIdentifier();
+    auto* blobData = m_blobs.get(urlKey);
+    if (m_isPartitioningEnabled && blobData && topOrigin && topOrigin != m_allowedBlobURLTopOrigins.get(urlKey)) {
```

Comment:
It seems check for blobData is unnecessary.

Reply:
I see your point. In this case, I think was thinking that the error message is only helpful if we have BlobData otherwise we'll just return `nullptr` in both cases. But, this situation could mean we have a bug somewhere, so adding the log message is a good idea.